services/web/server/tests/unit/with_dbs/functions_rpc/test_functions_controller_rpc.py

```diff
@@ -0,0 +1,493 @@
+# pylint: disable=redefined-outer-name
+from uuid import uuid4
+
+import pytest
+import simcore_service_webserver.functions._functions_controller_rpc as functions_rpc
+from models_library.api_schemas_webserver.functions_wb_schema import (
+    Function,
+    FunctionIDNotFoundError,
+    FunctionJobCollection,
+    FunctionJobIDNotFoundError,
+    JSONFunctionInputSchema,
+    JSONFunctionOutputSchema,
+    ProjectFunction,
+    ProjectFunctionJob,
+)
+
+
+@pytest.fixture
+def mock_function() -> Function:
+    return ProjectFunction(
+        uid=None,
+        title="Test Function",
+        description="A test function",
+        input_schema=JSONFunctionInputSchema(
+            schema_content={
+                "type": "object",
+                "properties": {"input1": {"type": "string"}},
+            }
+        ),
+        output_schema=JSONFunctionOutputSchema(
+            schema_content={
+                "type": "object",
+                "properties": {"output1": {"type": "string"}},
+            }
+        ),
+        project_id=uuid4(),
+        default_inputs=None,
+    )
+
+
+@pytest.mark.asyncio
+async def test_register_function(client, mock_function):
```

Comment:
can we have types for the parameterts?

Reply:
please address both of my suggestion in all the functions in this file

Reply:
Done